src/zenml/zen_server/routers/workspaces_endpoints.py

```diff
@@ -344,12 +344,25 @@ def create_stack(
 
         batch_verify_permissions_for_models(components, action=Action.READ)
 
-    return verify_permissions_and_create_entity(
+    needs_usage_increment = ResourceType.STACK in REPORTABLE_RESOURCES
+
+    if needs_usage_increment:
+        check_entitlement(ResourceType.STACK)
+
+    stack_response = verify_permissions_and_create_entity(
```

Comment:
This logic is already implemented in `verify_permissions_and_create_entity`, correct?